helix-view/src/handlers/lsp.rs

```diff
@@ -44,6 +44,17 @@ pub enum SignatureHelpEvent {
     RequestComplete { open: bool },
 }
 
+#[derive(Debug, PartialEq, Eq, Clone, Copy)]
+pub enum AutoSaveInvoked {
+    Automatic,
+}
+
+pub enum AutoSaveEvent {
+    // contains delay in milli seconds
+    Trigger(u64),
+    Cancel,
+}
```

Comment:
I don't think you need all of these types.

As far as I can tell the `AutoSaveEvent::Cancel` trigger is never invoked and `AutoSaveInvoked` is never constructed (and only has a single variant). I think the original PR had these types because it took the completions/signature help as reference.

You can simply set `type Event = u64;`.